### PR TITLE
Bump Jetpack submodule ref to fix node compatibility issue

### PR DESCRIPTION
**Pull request by guarani**
This brings in a Jetpack change that allows for node 14.17.6 to be used in the Jetpack submodule.

Related:
- https://github.com/Automattic/jetpack/pull/20990

### To test
- Verify that CI checks pass successfully including in optional tests

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

**Review by dcalhoun — APPROVED** [review]
LGTM. 🎉  CI jobs passing. I also ran the Demo app on an iPhone SE and Samsung Galaxy S20 with no issue. Thank you for putting this together! 

I suppose we will want to merge the Jetpack PR first and update the `jetpack` ref in this PR to be the merge commit before we merge this PR.

**Comment by guarani** [comment]
Thanks for your help here, @dcalhoun!

> I suppose we will want to merge the Jetpack PR first and update the `jetpack` ref in this PR to be the merge commit before we merge this PR.

Yes, we'll have to wait for the Jetpack PR to be merged before updating and merging this.

**Pull request merged**
